skimage/transform/histogram_matching.py

```diff
@@ -0,0 +1,72 @@
+import numpy as np
+
+
+def _match_cumulative_cdf(source, template):
+    """
+    Return modified source array so that the cumulative density function of
+    its values matches the cumulative density function of the template.
+    """
+    src_values, src_unique_indices, src_counts = np.unique(source.ravel(),
+                                                           return_inverse=True,
+                                                           return_counts=True)
+    tmpl_values, tmpl_counts = np.unique(template.ravel(), return_counts=True)
+
+    # calculate normalized quantiles for each array
+    src_quantiles = np.cumsum(src_counts).astype(np.float64)
+    src_quantiles /= src_quantiles[-1]
+
+    tmpl_quantiles = np.cumsum(tmpl_counts).astype(np.float64)
+    tmpl_quantiles /= tmpl_quantiles[-1]
+
+    interp_a_values = np.interp(src_quantiles, tmpl_quantiles, tmpl_values)
+    return interp_a_values[src_unique_indices].reshape(source.shape)
```

Comment:
This is a pretty damn nice implementation.

Reply:
Agreed; more elegant than any version I've seen.